dulwich/client.py

```diff
@@ -285,14 +284,14 @@ def read_pkt_refs_v2(
         ref = parts[1]
         for part in parts[2:]:
             if part.startswith(b"peeled:"):
-                peeled[ref] = part[7:]
+                refs[ref + PEELED_TAG_SUFFIX] = part[7:]
```

Comment:
please avoid deserializing and then reserializing; this also seems unnecessary.

Aren't the four lines around line 2680 all that we need?

Reply:
Something like those four lines, indeed. However there are two different implementations of get_refs() and we need to satisfy both of them. I have now force-pushed a smaller diff which doesn't have unnecessary changes.